compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ServiceDesugar.java

```diff
@@ -205,14 +206,20 @@ void rewriteServiceVariable(BLangService service, SymbolEnv env, BLangBlockStmt
             args.add(ASTBuilderUtil.createVariableRef(pos, service.serviceVariable.symbol));
 
             if (service.getServiceNameLiteral() == null) {
-                BLangListConstructorExpr.BLangArrayLiteral arrayLiteral =
-                        ASTBuilderUtil.createEmptyArrayLiteral(service.getPosition(), symTable.arrayStringType);
-                for (IdentifierNode path : service.getAbsolutePath()) {
-                    var literal = ASTBuilderUtil.createLiteral(path.getPosition(), symTable.stringType,
-                            path.getValue());
-                    arrayLiteral.exprs.add(literal);
+                if (service.getAbsolutePath().isEmpty()) {
```

Comment:
@rdhananjaya, hope this check here is sufficient.

Reply:
Change looks good to me